src/AutoRest.CSharp.V3/Plugins/Modeler.cs

```diff
@@ -243,30 +207,22 @@ private static ClientObjectConstant CreateConstant(Property property)
             return new ClientObjectConstant(property.CSharpName(), type, new ClientConstant(constantSchema.Value.Value, type));
         }
 
-        private static ClientObjectProperty CreateProperty(Property property)
-        {
-            return new ClientObjectProperty(property.CSharpName(), CreateType(property.Schema, property.IsNullable()), property.Schema.IsLazy(), property.SerializedName);
-        }
+        private static ClientObjectProperty CreateProperty(Property property) =>
+            new ClientObjectProperty(property.CSharpName(), CreateType(property.Schema, property.IsNullable()), property.Schema.IsLazy(), property.SerializedName);
 
-        private static ClientTypeReference CreateType(Schema schema, bool isNullable)
+        private static ClientTypeReference CreateType(Schema schema, bool isNullable) => schema switch
         {
-            switch (schema)
-            {
-                case BinarySchema _:
-                case ByteArraySchema _:
-                case Schema s when s.Type == AllSchemaTypes.Binary:
-                    return new BinaryTypeReference(false);
-                case ArraySchema array:
-                    return new CollectionTypeReference(CreateType(array.ElementType, false));
-                case DictionarySchema dictionary:
-                    return new DictionaryTypeReference(new FrameworkTypeReference(typeof(string)), CreateType(dictionary.ElementType, isNullable));
-                case Schema s when s.Type.ToFrameworkCSharpType() is Type type:
-                    return new FrameworkTypeReference(type, isNullable);
-                default:
-                    return new SchemaTypeReference(schema, isNullable);
-            }
-        }
-
+            BinarySchema _ => (ClientTypeReference)new BinaryTypeReference(false),
+            ByteArraySchema _ => new BinaryTypeReference(false),
+            { Type: AllSchemaTypes.Binary } => new BinaryTypeReference(false),
+            ArraySchema array => new CollectionTypeReference(CreateType(array.ElementType, false)),
+            DictionarySchema dictionary => new DictionaryTypeReference(new FrameworkTypeReference(typeof(string)), CreateType(dictionary.ElementType, isNullable)),
+            //https://devblogs.microsoft.com/dotnet/do-more-with-patterns-in-c-8-0/
+            { } s when s.Type.ToFrameworkCSharpType() is { } type => new FrameworkTypeReference(type, isNullable),
```

Comment:
I know resharper suggests this but do you think it's more readable than `Schema s when s.Type.ToFrameworkCSharpType() is Type`?

Reply:
Fixed.